### ISSUE TITLE
tool to expand scope or set of scopes

### ISSUE THREAD

**Issue by jonasfj** [taskcluster/taskcluster-tools#278]
Say I have a role like: `assume:repo:github.com/glandium/git-cinnabar:branch:master` and I want to know what scopes it gets... I need to call `auth.expandScopes` it would be nice to have a tool for this...

Note: `assume:repo:github.com/glandium/git-cinnabar:branch:<branch>` is basically a pattern made up by tc-gh...

**Comment by jonasfj** [comment]
This is just a nice-to-have quick hack.. Nothing important :)

**Comment by djmitche** [comment]
The pattern is actually documented as a Taskcluster convention in the namespaces document, and is also supported by mozilla-taskcluster.

**Comment by jonasfj** [comment]
I probably have no excuse for being too lazy to lookup the right docs...

But yes, namespaces document is where I should look.. still it's nice to be able to expand scopes.

**Comment by djmitche** [comment]
Yes, I was only disagreeing with your second paragraph.  I agree with the first :)

**Comment by Biboswan** [comment]
Can I work on this.

**Comment by djmitche** [comment]
Yes!  I think this would be an additional view which can take a list of scopes and expand it.

**Comment by Biboswan** [comment]
I 'm trying ```` await this.props.auth.expandScopes({ scopes: this.state.scopes }) ```` where scopes is an array

Getting error:
````
TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.
    at s (VM42094 290:2)
    at setTimeout (VM42094 290:2
````
https://docs.taskcluster.net/reference/platform/taskcluster-auth/references/api#expand-scopes

**Comment by djmitche** [comment]
Uh, wow.  So that API call isn't valid HTTP.  That was really stupid of me.

**Comment by djmitche** [comment]
So it sounds like you're on the right track!  Once the change linked above lands, we'll need to upgrade taskcluster-client-web in tools, and then the call you're making should work properly.

**Comment by Biboswan** [comment]
I see only taskcluster-client-web/src/clients/Auth.js needs an update.

**Comment by Biboswan** [comment]
Anything else?

**Comment by djmitche** [comment]
https://github.com/taskcluster/taskcluster-client-web#updating-built-in-apis shows how to update the client and make a PR.  I'll try to get to that today, but if you beat me to it and make a PR that's great too :)

**Comment by djmitche** [comment]
OK, if you `yarn add taskcluster-client-web@5.1.0` you should get a working
expandScopes.

**Comment by eliperelman** [comment]
You can shorten that to `yarn add taskcluster-client-web`.

**Comment by Biboswan** [comment]
Now I'm getting expandScopes response successfully.:)
